File: backend/src/SportsGurukul.Application/Features/AcademyManagement/DTOs/AcademyDto.cs
namespace SportsGurukul.Application.Features.AcademyManagement.DTOs;

public class AcademyDto
{
    public Guid Id { get; set; }
    public string AcademyCode { get; set; } = string.Empty;
    public string Name { get; set; } = string.Empty;
    public string? LegalName { get; set; }
    public string? Description { get; set; }
    public string? RegistrationNumber { get; set; }
    public string? GSTNumber { get; set; }
    public DateTime? EstablishedDate { get; set; }
    public string? Website { get; set; }
    public string Email { get; set; } = string.Empty;
    public string Phone { get; set; } = string.Empty;
    public string Status { get; set; } = string.Empty;
    public string VerificationStatus { get; set; } = string.Empty;
    public string? LogoUrl { get; set; }
    public string? BannerUrl { get; set; }
    public ContactDto? Contact { get; set; }
    public OperatingHoursDto? OperatingHours { get; set; }
    public IReadOnlyList<BranchDto> Branches { get; set; } = [];
    public IReadOnlyList<AcademySportDto> Sports { get; set; } = [];
    public IReadOnlyList<FacilityDto> Facilities { get; set; } = [];
    public IReadOnlyList<MembershipPlanDto> Memberships { get; set; } = [];
    public IReadOnlyList<SocialLinkDto> SocialLinks { get; set; } = [];
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

File: mobile/lib/features/academy/create/infrastructure/models/academy_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:sports_gurukul/features/academy/create/infrastructure/models/academy_branch_dto.dart';
import 'package:sports_gurukul/features/academy/create/infrastructure/models/academy_contact_dto.dart';
import 'package:sports_gurukul/features/academy/create/infrastructure/models/academy_sport_dto.dart';

part 'academy_dto.freezed.dart';
part 'academy_dto.g.dart';

/// DTO matching the backend `AcademyDto` schema returned by
/// `POST /api/v1/academies` and the academy GET endpoints.
@freezed
abstract class AcademyDto with _$AcademyDto {
  const factory AcademyDto({
    required String id,
    required String academyCode,
    required String name,
    required String email,
    required String phone,
    required String status,
    required String verificationStatus,
    required String createdAt,
    String? legalName,
    String? description,
    String? registrationNumber,
    String? gstNumber,
    String? establishedDate,
    String? website,
    String? academyType,
    String? logoUrl,
    String? bannerUrl,
    String? updatedAt,
    AcademyContactDto? contact,
    List<AcademyBranchDto>? branches,
    List<AcademySportDto>? sports,
  }) = _AcademyDto;

  factory AcademyDto.fromJson(Map<String, dynamic> json) =>
      _$AcademyDtoFromJson(json);
}
